plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMinioConnectorSmokeTest.java

```diff
@@ -138,8 +143,65 @@ public void testMetadataLocationWithDoubleSlash()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testExpireSnapshotsBatchDeletes()
+    {
+        String tableName = "test_expiring_snapshots_" + randomTableSuffix();
+        Session sessionWithShortRetentionUnlocked = prepareCleanUpSession();
+        String location = "s3://%s/%s/%s/".formatted(bucketName, schemaName, tableName);
+        List<Event> events = new ArrayList<>();
+        hiveMinioDataLake.getMinioClient().captureBucketNotifications(bucketName, event -> {
+            if (event.eventType().toString().toLowerCase(ENGLISH).contains("remove")) {
+                events.add(event);
+            }
+        });
+
+        assertUpdate("CREATE TABLE " + tableName + " (key varchar, value integer) WITH (location='" + location + "')");
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('one', 1)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('two', 2)", 1);
+        assertThat(query("SELECT * FROM " + tableName)).matches("VALUES (VARCHAR 'one', 1), (VARCHAR 'two', 2)");
+
+        List<String> initialMetadataFiles = hiveMinioDataLake.getMinioClient().listObjects(bucketName, "/%s/%s/metadata".formatted(schemaName, tableName));
+        assertThat(initialMetadataFiles).isNotEmpty();
+
+        List<Long> initialSnapshots = getSnapshotIds(tableName);
+        assertThat(initialSnapshots).hasSizeGreaterThan(1);
+
+        assertQuerySucceeds(sessionWithShortRetentionUnlocked, "ALTER TABLE " + tableName + " EXECUTE EXPIRE_SNAPSHOTS (retention_threshold => '0s')");
+
+        List<String> updatedMetadataFiles = hiveMinioDataLake.getMinioClient().listObjects(bucketName, "/%s/%s/metadata".formatted(schemaName, tableName));
+        assertThat(updatedMetadataFiles).isNotEmpty().hasSizeLessThan(initialMetadataFiles.size());
+
+        List<Long> updatedSnapshots = getSnapshotIds(tableName);
+        assertThat(updatedSnapshots).hasSize(1);
+
+        assertThat(query("SELECT * FROM " + tableName))
+                .matches("VALUES (VARCHAR 'one', 1), (VARCHAR 'two', 2)");
+        // if files were deleted in batch there should be only one request id because there was one request only
+        assertThat(events.stream()
+                .map(event -> event.responseElements().get("x-amz-request-id"))
+                .collect(toImmutableSet())).hasSize(1);
```

Comment:
Can we also verify that more than 1 file got deleted?

if it is already being asserted, i didn't notice, sorry